include/boost/algorithm/cxx11/move_if.hpp

```diff
@@ -0,0 +1,96 @@
+/* 
+   Copyright (c) Denis Mikhailov 2022.
+
+   Distributed under the Boost Software License, Version 1.0. (See accompanying
+   file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+*/
+
+/// \file  move_if.hpp
+/// \brief Move a subset of a sequence to a new sequence
+/// \author Denis Mikhailov
+
+#ifndef BOOST_ALGORITHM_MOVE_IF_HPP
+#define BOOST_ALGORITHM_MOVE_IF_HPP
+
+#include <utility>    // for std::pair, std::make_pair, std::move
+
+#include <boost/config.hpp>
+#include <boost/range/begin.hpp>
+#include <boost/range/end.hpp>
+
+namespace boost { namespace algorithm {
+
+/// \fn move_while ( InputIterator first, InputIterator last, OutputIterator d_first, Predicate p )
+/// \brief Moves all the elements at the start of the range that
+///     satisfy the predicate to the another range.
+/// \return Updated two iterators for the first and the second range
+/// 
+/// \param first    The start of the range to move
+/// \param last     One past the end of the range to move
+/// \param d_first  The start of the destination range
+/// \param p        A predicate for testing the elements of the range
+/// \note           C++11-compatible compiler required.
+template<typename InputIterator, typename OutputIterator, typename Predicate> 
```

Comment:
rename concept's name. `InputIterator` is not enough here

Reply:
It's not a truth, implementation of `std::move` algorithm from libstdc++ says that `InputIterator` is enough here:
https://github.com/gcc-mirror/gcc/blob/releases/gcc-12.2.0/libstdc++-v3/include/bits/stl_algobase.h#L647

Reply:
Conversation was resolved without any changes